// @entur/component-library/lib/components/expandable/expandableHeader/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ExpandableHeader = ExpandableHeader;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _chevron = require('../../icons/arrows/chevron');

var _chevron2 = _interopRequireDefault(_chevron);

require('./styles.css');

var _styles = {
  'rootClassName': 'entur-expandable-header',
  'entur-expandable-header': 'entur-expandable-header',
  'entur-expandable-header__arrow': 'entur-expandable-header__arrow',
  'disabled': 'disabled',
  'entur-expandable-header--variant-midnight': 'entur-expandable-header--variant-midnight',
  'expanded': 'expanded',
  'entur-expandable': 'entur-expandable'
};

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function ExpandableHeader(_ref) {
  var className = _ref.className,
      children = _ref.children,
      expanded = _ref.expanded,
      hideArrowIcon = _ref.hideArrowIcon,
      onClick = _ref.onClick,
      variant = _ref.variant;

  var classNames = (0, _classnames2.default)(_styles.rootClassName, className, _defineProperty({}, _styles.rootClassName + '--variant-' + variant, variant));
  var arrowDirection = expanded ? 'up' : 'down';
  return _react2.default.createElement(
    'div',
    { className: classNames, onClick: onClick },
    children,
    !hideArrowIcon && _react2.default.createElement(_chevron2.default, {
      className: _styles.rootClassName + '__arrow',
      direction: arrowDirection
    })
  );
}

ExpandableHeader.propTypes = {
  children: _propTypes2.default.node.isRequired,
  className: _propTypes2.default.string,
  expanded: _propTypes2.default.bool,
  hideArrowIcon: _propTypes2.default.bool,
  onClick: _propTypes2.default.func,
  variant: _propTypes2.default.oneOf(['primary', 'midnight'])
};
ExpandableHeader.__docgenInfo = {
  'description': '',
  'methods': [],
  'displayName': 'ExpandableHeader',
  'props': {
    'children': {
      'type': {
        'name': 'node'
      },
      'required': true,
      'description': ''
    },
    'className': {
      'type': {
        'name': 'string'
      },
      'required': false,
      'description': ''
    },
    'expanded': {
      'type': {
        'name': 'bool'
      },
      'required': false,
      'description': ''
    },
    'hideArrowIcon': {
      'type': {
        'name': 'bool'
      },
      'required': false,
      'description': ''
    },
    'onClick': {
      'type': {
        'name': 'func'
      },
      'required': false,
      'description': ''
    },
    'variant': {
      'type': {
        'name': 'enum',
        'value': [{
          'value': '\'primary\'',
          'computed': false
        }, {
          'value': '\'midnight\'',
          'computed': false
        }]
      },
      'required': false,
      'description': ''
    }
  }
};
