// @entur/component-library/src/components/legLine/index.js
// @flow
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { rootClassName } from './styles.scss';

export function LegLine({
  className,
  direction = 'horizontal',
  pattern,
  color,
  endColor,
  startColor,
  showStart = true,
  showLine = true,
  showStop = true,
}) {

  const classNames = cx(rootClassName, {
    [`${rootClassName}--direction-${direction}`]: direction,
  }, className);

  const middleClassName = cx(`${rootClassName}__middle`, {
    [`${rootClassName}__middle--pattern-${pattern}`]: pattern,
  });

  return (
    <div className={ classNames }>
      { showStart &&
        <div
          className={`${rootClassName}__start`}
          style={ { backgroundColor: startColor || color } }
        />
      }

      { showLine &&
        <div
          className={ middleClassName }
          style={ { backgroundColor: color } }
        />
      }

      { showStop &&
        <div
          className={`${rootClassName}__stop`}
          style={ { backgroundColor: endColor || color } }
        />
      }
    </div>
  );
}

LegLine.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
  endColor: PropTypes.string,
  pattern: PropTypes.oneOf(['line', 'dashed', 'dotted', 'wave']),
  startColor: PropTypes.string,
  showStart: PropTypes.bool,
  showLine: PropTypes.bool,
  showStop: PropTypes.bool,
};
