// @entur/component-library/lib/components/loading/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Loading = Loading;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./styles.css');

var _styles = {
  'rootClassNameDetermined': 'entur-loader-determined',
  'rootClassNameUndetermined': 'entur-loader-undetermined',
  'entur-loader-undetermined__label': 'entur-loader-undetermined__label',
  'entur-loader-undetermined__loader': 'entur-loader-undetermined__loader',
  'entur-loader-undetermined__indicator': 'entur-loader-undetermined__indicator',
  'entur-loader-undetermined--slow-animation': 'entur-loader-undetermined--slow-animation',
  'back-and-forth': 'back-and-forth',
  'entur-loader-undetermined--fast-animation': 'entur-loader-undetermined--fast-animation',
  'entur-loader-determined__label': 'entur-loader-determined__label',
  'entur-loader-determined__progress-bar': 'entur-loader-determined__progress-bar',
  'entur-loader-determined__indicator': 'entur-loader-determined__indicator',
  'entur-loader-determined__progress': 'entur-loader-determined__progress'
};

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Loading(_ref) {
  var label = _ref.label,
      progress = _ref.progress,
      hideProgress = _ref.hideProgress,
      animationSpeed = _ref.animationSpeed,
      color = _ref.color;

  var undeterminedIndicatorClassNames = (0, _classnames2.default)(_styles.rootClassNameUndetermined + '__indicator', _styles.rootClassNameUndetermined + '--' + animationSpeed + '-animation');

  return progress != null ? _react2.default.createElement(
    'div',
    { className: '' + _styles.rootClassNameDetermined },
    _react2.default.createElement(
      'span',
      { className: _styles.rootClassNameDetermined + '__label' },
      label
    ),
    _react2.default.createElement(
      'div',
      { className: _styles.rootClassNameDetermined + '__progress-bar' },
      _react2.default.createElement('span', {
        className: _styles.rootClassNameDetermined + '__indicator',
        style: { width: progress + '%', backgroundColor: color }
      })
    ),
    !hideProgress && _react2.default.createElement(
      'span',
      { className: _styles.rootClassNameDetermined + '__progress' },
      progress + '%'
    )
  ) : _react2.default.createElement(
    'div',
    { className: '' + _styles.rootClassNameUndetermined },
    _react2.default.createElement(
      'span',
      { className: _styles.rootClassNameUndetermined + '__label' },
      label
    ),
    _react2.default.createElement(
      'div',
      { className: _styles.rootClassNameUndetermined + '__loader' },
      _react2.default.createElement('span', {
        className: undeterminedIndicatorClassNames,
        style: { backgroundColor: color }
      })
    )
  );
}

Loading.propTypes = {
  label: _propTypes2.default.string,
  progress: _propTypes2.default.number,
  hideProgress: _propTypes2.default.bool,
  animationSpeed: _propTypes2.default.oneOf(['slow', 'fast']),
  color: _propTypes2.default.string
};

Loading.defaultProps = {
  label: '',
  hideProgress: false,
  animationSpeed: 'slow',
  color: '#ff5959'
};
Loading.__docgenInfo = {
  'description': '',
  'methods': [],
  'displayName': 'Loading',
  'props': {
    'label': {
      'defaultValue': {
        'value': '\'\'',
        'computed': false
      },
      'type': {
        'name': 'string'
      },
      'required': false,
      'description': ''
    },
    'hideProgress': {
      'defaultValue': {
        'value': 'false',
        'computed': false
      },
      'type': {
        'name': 'bool'
      },
      'required': false,
      'description': ''
    },
    'animationSpeed': {
      'defaultValue': {
        'value': '\'slow\'',
        'computed': false
      },
      'type': {
        'name': 'enum',
        'value': [{
          'value': '\'slow\'',
          'computed': false
        }, {
          'value': '\'fast\'',
          'computed': false
        }]
      },
      'required': false,
      'description': ''
    },
    'color': {
      'defaultValue': {
        'value': '\'#ff5959\'',
        'computed': false
      },
      'type': {
        'name': 'string'
      },
      'required': false,
      'description': ''
    },
    'progress': {
      'type': {
        'name': 'number'
      },
      'required': false,
      'description': ''
    }
  }
};
