// @entur/component-library/src/components/slideSwitch/index.js
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { rootClassName } from './styles.scss';
import { Label } from '../label';

export function SlideSwitch(props) {
  const {
    id,
    value,
    checked,
    className,
    onChange,
    label,
    disabled,
    color,
    size,
    knobIcon,
    ...rest
  } = props;

  const combinedClassNames = cx(rootClassName, {
    [`${rootClassName}--size-${size}`]: size,
  }, className);

  const checkboxClassName = `${rootClassName}__input`;
  const indicatorClassName = `${rootClassName}__indicator`;
  const labelClassName = `${rootClassName}__label`;
  const knobClassName = `${indicatorClassName}__knob`;

  const indicatorStyle = checked && !disabled ? { backgroundColor: color, borderColor: color } : undefined;
  const knobStyle = checked && !disabled ? { borderColor: color } : undefined;

  return (
    <div className={combinedClassNames}>
      <input
        id={id}
        className={checkboxClassName}
        type="checkbox"
        value={value}
        checked={checked}
        onChange={onChange}
        disabled={disabled}
        {...rest}
      />
      <div className={indicatorClassName} style={ indicatorStyle }>
        <div className={knobClassName} style={ knobStyle }>
          { knobIcon }
        </div>
      </div>
      <Label htmlFor={id} className={labelClassName}>
        {label}
      </Label>
    </div>
  );
}

SlideSwitch.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  checked: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  label: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.oneOf(['medium', 'large']),
  knobIcon: PropTypes.node,
};

SlideSwitch.defaultProps = {
  checked: false,
  value: '',
  onChange: () => {},
  disabled: false,
  label: '',
  size: 'medium',
};
